packages/server/src/threads/query.js

```diff
@@ -189,6 +183,11 @@ class QueryRunner {
     if (!resp.error) {
       const globalUserId = getGlobalIDFromUserMetadataID(_id)
       await updateUserOAuth(globalUserId, resp)
+      this.ctx.user = await userCache.getUser(globalUserId)
+    } else {
+      // In this event the user may have oAuth issues that
+      // could require re-authenticating with their provider.
+      throw new Error("OAuth2 access token could not be refreshed")
```

Comment:
Probably worth adding in the `resp.error` or `resp.error.message` (not sure if error is an `Error` or a string) here so that we have some insight into what went wrong with the provider